errors/dynamic-server-error.mdx

```diff
@@ -0,0 +1,63 @@
+---
+title: DynamicServerError - Dynamic Server Usage
+---
+
+#### Why This Message Occurred
+
+You attempted to use a Next.js function that depends on Async Context (such as `headers` or `cookies` from `next/headers`) but it was not bound to the same call stack as the function that ran it. Next.js will throw a `DynamicServerError` to know when to opt a page into dynamic rendering. However, when it's uncaught, that means that the async context was lost.
```

Comment:
How would I know that it's bound to the same call stack? This feels like an implementation detail. What user code causes this?